File: app/src/main/java/com/suzei/minote/ui/settings/SettingsContract.java
package com.suzei.minote.ui.settings;

import android.content.Intent;

import com.suzei.minote.utils.ColorWheel;

public interface SettingsContract {

    interface View {

        void setPresenter(Presenter presenter);

        void showColorWheel(String title, String initialColor, ColorWheel colorWheel);

        void startIntentActivity(Intent intent);

    }

    interface Presenter {

        void noteColorWheel();

        void textColorWheel();

        void redirectToEmail();

        void redirectToPlaystore();

    }

}

File: app/src/main/java/com/suzei/minote/ui/settings/SettingsContract.kt
package com.suzei.minote.ui.settings

interface SettingsContract {

    interface View {

        fun setPresenter(presenter: Presenter)

        fun setDetails(isCheck: Boolean)

        fun showToastMessage(message: String)

    }

    interface Presenter {

        fun setup()

        fun setAutoSave(isCheck: Boolean)

    }

}